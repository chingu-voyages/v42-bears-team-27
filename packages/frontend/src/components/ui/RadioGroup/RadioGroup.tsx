import type { ForwardRefExoticComponent, RefAttributes } from 'react';
import { forwardRef } from 'react';
import type { ThemeUIStyleObject } from 'theme-ui';
import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';

export interface RadioGroupProps extends RadioGroupPrimitive.RadioGroupProps {
  sx?: ThemeUIStyleObject;
}

export const RadioGroup: ForwardRefExoticComponent<
RadioGroupProps & RefAttributes<HTMLDivElement>
> = forwardRef(({ children, sx, ...rest }, ref) => (
  <RadioGroupPrimitive.Root
    ref={ref}
    sx={{ display: 'flex', flexDirection: 'column', rowGap: 2, ...sx }}
    {...rest}
  >
    {children}
  </RadioGroupPrimitive.Root>
));

if (process.env.NODE_ENV !== 'production') {
  RadioGroup.displayName = 'RadioGroup';
}

export interface RadioProps extends RadioGroupPrimitive.RadioGroupItemProps {
  /**
   * Label for the component
   */
  label: string;
  sx?: ThemeUIStyleObject;
}

export const Radio: ForwardRefExoticComponent<
RadioProps & RefAttributes<HTMLButtonElement>
> = forwardRef(({ id, label, sx, ...rest }, ref) => (
  <div
    key={id}
    sx={{ display: 'flex', alignItems: 'center', columnGap: 3, ...sx }}
  >
    <RadioGroupPrimitive.Item
      ref={ref}
      id={id}
      sx={{
        variant: 'buttons.primary.outlined',
        width: 24,
        height: 24,
        borderRadius: '100%',
        cursor: 'pointer',
      }}
      {...rest}
    >
      <RadioGroupPrimitive.Indicator
        sx={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
          width: '100%',
          height: '100%',
          position: 'relative',
          '&::after': {
            content: '""',
            position: 'absolute',
            width: 12,
            height: 12,
            borderRadius: '50%',
            bg: 'primary',
          },
        }}
      />
    </RadioGroupPrimitive.Item>
    <label sx={{ variant: 'text.label' }} htmlFor={id}>
      {label}
    </label>
  </div>
));

if (process.env.NODE_ENV !== 'production') {
  Radio.displayName = 'Radio';
}
