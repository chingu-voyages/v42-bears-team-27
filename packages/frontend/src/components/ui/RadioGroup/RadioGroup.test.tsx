import { render } from 'test-utils';

import { RadioGroup, Radio } from './RadioGroup';

describe('<RadioGroup />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <RadioGroup>
          <Radio value="foo" label="Foo" />
          <Radio value="bar" label="Bar" />
          <Radio value="baz" label="Baz" />
        </RadioGroup>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
