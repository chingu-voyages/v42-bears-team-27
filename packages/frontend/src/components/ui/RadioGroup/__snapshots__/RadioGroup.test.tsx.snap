// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> Snapshots should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 16px;
  column-gap: 16px;
}

.emotion-2 {
  background-color: white;
  color: var(--theme-ui-colors-primary);
  border: 2px solid currentColor;
  width: 24px;
  height: 24px;
  border-radius: 100%;
  cursor: pointer;
}

.emotion-2:hover {
  background-color: #e6e6e6;
}

.emotion-3 {
  font-family: "Nunito",sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  color: var(--theme-ui-colors-primary);
}

<div>
  <div
    aria-required="false"
    class="emotion-0"
    dir="ltr"
    role="radiogroup"
    style="outline: none;"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <button
        aria-checked="false"
        class="emotion-2"
        data-radix-collection-item=""
        data-state="unchecked"
        role="radio"
        tabindex="-1"
        type="button"
        value="foo"
      />
      <label
        class="emotion-3"
      >
        Foo
      </label>
    </div>
    <div
      class="emotion-1"
    >
      <button
        aria-checked="false"
        class="emotion-2"
        data-radix-collection-item=""
        data-state="unchecked"
        role="radio"
        tabindex="-1"
        type="button"
        value="bar"
      />
      <label
        class="emotion-3"
      >
        Bar
      </label>
    </div>
    <div
      class="emotion-1"
    >
      <button
        aria-checked="false"
        class="emotion-2"
        data-radix-collection-item=""
        data-state="unchecked"
        role="radio"
        tabindex="-1"
        type="button"
        value="baz"
      />
      <label
        class="emotion-3"
      >
        Baz
      </label>
    </div>
  </div>
</div>
`;
